### PR TITLE
fixes for time partition, schema changes, and so

### PR DESCRIPTION
**Pull request by dorinhogea**
Resubmit after rebase.

### CODE REVIEW

File: db/views.h [hunk 1]
```diff
@@ -261,12 +270,27 @@ int timepart_is_shard(const char *name, int lock);
 int timepart_is_timepart(const char *name, int lock);
 
 /**
- * Alter a timepart 
+ * Time partition schema change resume
+ *
+ */
+int timepart_resume_schemachange(int check_llmeta(const char *));
```
Comment (ldennis) [inline db/views.h]: I don't see this function being called. Why do we need this?

File: db/views.h [hunk 2]
```diff
@@ -261,12 +270,27 @@ int timepart_is_shard(const char *name, int lock);
...
+
+/**
+ * During resume, we need to check at if the interrupted alter made any
+ * progress, and continue with that shard
+ *
+ */
+int timepart_schemachange_get_shard_in_progress(const char *view_name,
```
Comment (ldennis) [inline db/views.h]: same as above. Why do we need this? I think `resume_schema_change` already loops through all tables in the `thedb->num_dbs` array.

**Comment by ldennis** [comment]
Does this also resolve timepart sc resume issue?
And does this fix races you found for views on replicants?

**Comment by dorinhogea** [comment]
No, the resume will need review and another PR.

File: db/views.h (hunk 2 above)
Comment (dorinhogea) [reply db/views.h]: I need these for partition resume, it is not done.

**Comment by dorinhogea** [comment]
This PT is just meant to refactor time partition schema changes (i.e. eliminate lower level handling) , and align the code with the new refactored tran ddl changes.

File: db/views.h (hunk 2 above)
Comment (ldennis) [reply db/views.h]: I think resume_schema_change already takes care of resuming multiple schema changes?

**Comment by ldennis** [comment]
The refactored code looks good.

**Comment by dorinhogea** [comment]
thank you

**Pull request merged**